Allow files called just test.jsx? or spec.jsx? in root

SimenB opened the pull request:
**Summary**

I have a file called `test.js` in the root of my project, and I think jests should automatically pick it up without me providing custom `testRegex`.

**Test plan**

```sh-session
$ node_modules/.bin/jest
 PASS  ./test.js
  ✓ assets should always be an array (2ms)
  ✓ assets should should be reversed (1ms)
  ✓ should invoke callback on success (7ms)
  ✓ should invoke callback on error (1ms)
  ✓ should add file using compilation's publicPath (3ms)
  ✓ should used passed in publicPath
  ○ should handle missing `publicPath`
  ✓ should add file missing "/" to public path
  ✓ should add sourcemap to compilation
  ✓ should skip adding sourcemap to compilation if set to false
  ✓ should include hash of file content if option is set
  ✓ should add to css if `typeOfAsset` is css
  ✓ should replace compilation assets key if `outputPath` is set

Test Suites: 1 passed, 1 total
Tests:       1 skipped, 12 passed, 13 total
Snapshots:   0 total
Time:        1.084s
Ran all test suites.
```

I tried writing a test for it, but there's no tests actually testing the default regex, AFAICT.

cpojer commented:
There should be some tests for this in integration tests.

I'm not sure if we should go ahead with this. If you only have one test file, you can adjust the testRegex in your configuration to read these files. I'd rather not make the name of test files more ambiguous.

SimenB commented:
It's a common pattern in small modules (which normally has just an `index.js` and a `test.js` in the root of the project), both Mocha and Ava supports it out of the box

cpojer commented:
However, with your change "Footest.js" or "Myspec.js" will now also be considered as tests.

SimenB commented:
Good point, updated.

How would I add an integration tests? I'm unsure how to add say a `test.js` and a `notatest.js` and assert that only one was picked up.

cpojer commented:
I think we also have a bunch of places (docs?) where this regex is used.

We use Jest for integration tests. Those tests invoke Jest, so there is some jestception. Simply check out the `integration_tests` folder. Every folder comes with its own Jest config and the files in `integration_tests/__tests__` are the ones that run Jest in one of those folders and make sure the CLI reports the correct output.

SimenB commented:
Updated the docs, and added an integration test that seems to do what I want. I made the assertions in the files I don't want to test fail, but I'm unsure how I can ensure that the ones who should get picked up are?

cpojer commented:
Nice! See `integration_tests/__tests__` and check out some of the existing tests there and what they check there – you need to add one of those to make sure your integration test folder will actually be run in our suite.

cpojer commented:
So basically when you do `jest integration_tests/__tests__/my-test.js`, those tests are expected to invoke Jest itself in `integration_tests/<my-test-folder>/` and then make sure the output is what we expect.

SimenB commented:
Ah, gotcha! Added a test for it now. When I got it, the tests made a lot of sense 😄

cpojer commented on integration_tests/__tests__/test-in-root-test.js:
```diff
@@ -0,0 +1,28 @@
+const runJest = require('../runJest');
+
+it('runs tests in only test.js and spec.js', () => {
+  const result = runJest('test-in-root');
```
would you mind checking the other tests in the `integration_test/__tests__` folder? We can run jest with the json flag and then have a JS object instead of using regex – which is going to be harder to maintain :)

SimenB replied:
Changing to `const result = runJest.json('test-in-root');` makes no difference, the test still passes. Is that enough?

SimenB replied:
Ah, then `.json`. That's much better, thanks!

SimenB commented:
I don't get the test failure, but it's failing on master as well.

SimenB commented:
@cpojer is this ok now?

cpojer commented:
yep, looks good!
